# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1218552871 14400
#      Tue Aug 12 10:54:31 2008 -0400
# Branch release-3-0-x
# Node ID daf91931b32bd50eb8a3e7cbe1bf9cfd23f44c1c
# Parent  b7f2ab4e34b0c60a08af70a1f6ab2700717d2e8f
initialization check for correct NaN sign

diff --git a/liboctave/ChangeLog b/liboctave/ChangeLog
--- a/liboctave/ChangeLog
+++ b/liboctave/ChangeLog
@@ -1,8 +1,13 @@
+2008-08-12  Jaroslav Hajek <highegg@gmail.com>
+
+	* lo-ieee.cc (octave_ieee_init): Try to ensure that octave_NaN is
+	classified as positive by lo_ieee_signbit.
+
 2008-08-04  John W. Eaton  <jwe@octave.org>
 
 	* oct-env.cc (octave_env::instance_ok): Fix typo in error message.
 
 2008-06-20  Jaroslav Hajek <highegg@gmail.com>
 
 	* MatrixType.h: Add missing include statement.
 
diff --git a/liboctave/lo-ieee.cc b/liboctave/lo-ieee.cc
--- a/liboctave/lo-ieee.cc
+++ b/liboctave/lo-ieee.cc
@@ -89,16 +89,20 @@ octave_ieee_init (void)
 	  }
 #endif
 
 #if defined (__alpha__) && defined (__osf__)
 	extern unsigned int DQNAN[2];
 	octave_NaN = (*(X_CAST(double *, DQNAN)));
 #else
 	octave_NaN = tmp_inf / tmp_inf;
+        // try to ensure that lo_ieee_sign gives false for a NaN.
+        if (lo_ieee_signbit (octave_NaN))
+          octave_NaN = -octave_NaN;
+
 #endif
 
 	octave_Inf = tmp_inf;
 
 	// This is patterned after code in R.
 
 	if (ff == oct_mach_info::flt_fmt_ieee_big_endian)
 	  {
